pkg/utils/statsutils/metrics/metrics.go

```diff
@@ -60,21 +63,33 @@ func GetDefaultConfigStatusOptions() map[string]*MetricLabels {
 // NewConfigStatusMetrics creates and returns a ConfigStatusMetrics from the specified options.
 // If the options are invalid, an error is returned.
 func NewConfigStatusMetrics(opts map[string]*MetricLabels) (ConfigStatusMetrics, error) {
+	metrics, err := prepareMetrics(opts)
+	if err != nil {
+		return ConfigStatusMetrics{}, err
+	}
+
 	configMetrics := ConfigStatusMetrics{
-		metrics: make(map[schema.GroupVersionKind]*resourceMetric),
+		opts:    opts,
+		metrics: metrics,
 	}
+
+	return configMetrics, nil
+}
+
+func prepareMetrics(opts map[string]*MetricLabels) (map[schema.GroupVersionKind]*resourceMetric, error) {
+	metrics := make(map[schema.GroupVersionKind]*resourceMetric)
 	for gvkString, labels := range opts {
 		gvk, err := parseGroupVersionKind(gvkString)
 		if err != nil {
-			return ConfigStatusMetrics{}, err
+			return metrics, err
 		}
 		metric, err := newResourceMetric(gvk, labels.GetLabelToPath())
 		if err != nil {
-			return ConfigStatusMetrics{}, err
+			return metrics, err
```

Comment:
```suggestion
			return map[schema.GroupVersionKind]*resourceMetric{}, err
```
Same as above lets return an empty map